tools/restart_harvester.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+# Custom script to ensure harvester processes are safe to restart
+# Based on
+#  - https://github.com/GSA/data.gov/issues/3796
+#  - https://github.com/GSA/data.gov/issues/3962
+
+# Input either catalog-gather or catalog-fetch
+app_to_restart=$1
+
+###################
+# Utility Functions
+# Get the number of lines in the log (int)
+log_count () { cf logs --recent $app_to_restart | wc -l; }
+
+# Get the time of the last line in the log (int)
+current_time=$(date --utc +%s)
+log_last_time () {
+  date --utc --date="$(cf logs --recent $app_to_restart | tail -n 1 | awk '{split($0,time," "); print time[1]}')" +%s
+}
+
+# Restart if CPU usage is < 1
+cpu_restart () {
+  instances=$(cf app $app_to_restart | grep '^instances:' | sed 's/.*\///')
+  i=0
+  cf app $app_to_restart | tail -n $instances | awk '{ split($4,cpu,"."); print cpu[1]}' | while read -r cpu ; do
+    if [[ $cpu < 1 ]]; then
```

Comment:
where do we get `$cpu` from?

Reply:
The `while read -r cpu` iterates through the previous command and makes the iteration available through the `cpu` variable.

Reply:
Thanks! I feel like this line might benefit from a comment explaining the pipes.